### ISSUE TITLE
AOB scan doesn't seem to work well on 64 bit address.

### ISSUE THREAD

**Issue by errorc0de404** [erfg12/memory.dll#10]
![help](https://user-images.githubusercontent.com/35400520/34906820-d6d5e592-f86c-11e7-8c11-f4680dd2d4c8.JPG)
I wanted to scan this array of bytes on address '1AAAA2EE82C'

![help2](https://user-images.githubusercontent.com/35400520/34906822-da18222e-f86c-11e7-9c17-3843433d0811.JPG)
So I set the start address to 0x50000000 and range to 0x30000000000.
If it works well, it should search 0x50000000-0x30050000000.

![help3](https://user-images.githubusercontent.com/35400520/34906823-db7f53a8-f86c-11e7-8786-035f43cfda1a.JPG)
 But when I checked debug strings, the range that I set and the range that it actually scanned was diffrent.
(The scan worked well when I tried same thing with 32-bit address.) 

![help4](https://user-images.githubusercontent.com/35400520/34907001-a9a20b48-f86f-11e7-81f0-ea874c75b5de.JPG)
I even unchecked the 'prefer 32bit' on build setting.
How do I get it to scan the same range as I entered?

**Comment by hydra-chan** [comment]
Hi, I was actually doing a little bit of research on ReadProcessMemory, 64 bit Windows, 64 bit apps, etc.

Now what I did was besides uncheck "Prefer 32-bit", and given my scope and expectations for my current work, is set the target platform to be 64 bit in the dropdown menu above the checkbox you've indicated.

Then, I followed this [solution](https://stackoverflow.com/questions/34467311/c-sharp-readprocessmemory-how-to-read-a-64-bit-memory-address)
and this [information](https://stackoverflow.com/questions/1364741/does-64-bit-windows-use-kernel64).

Now, the answer in the first link is provided by bogza.anton and I would like to note that rather than replacing `int lpBaseAddress` with `Int64 lpBaseAddress`, it might be reasonable to actually use a `UInt64` instead (since memory addresses shouldn't go negative).

Basically, if you're using 64 bit Windows, importing from kernel32 will give you functions that should be compatible with 64 bit programs. You just need to change the imported function's signature.

In this way, I was able to get a dll import to work properly with my 64 bit app to read a 64 bit game's memory.

I predict that this solution would also be extendible to this project as whole, such that if you modified the source from `int` address parameters to `UInt64` address parameters and recompiled the library, there would be little to no issue. Haven't tried it myself quite yet, but my excitement for this solution and initial results prompts me to post before I forget.

**Comment by erfg12** [comment]
Fork the source, make changes, and send me a push request.

**Comment by erfg12** [comment]
Just merged hollow87's aobscan improvements. Try them out.

**Comment by erfg12** [comment]
Closing issue because errorc0de404 isn't responding.
